src/basics.md

```diff
@@ -362,6 +363,52 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-verify-extract-email]: #ex-verify-extract-email
+<a name="ex-verify-extract-email"></a>
+## Verify and extract login from an email address
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Validates that an email address is formatted correctly, and extracts everything
+before the @ symbol.
+
+```rust
+#[macro_use]
+extern crate lazy_static;
+extern crate regex;
+
+use regex::Regex;
+use std::borrow::Cow;
+
+// Validates that something has an @ symbol, followed by something that
+// looks like a domain. Returns what came before the @ symbol, if everything
```

Comment:
i would pare the comments down as much as possible or even remove them (the example should be self explanatory as much as possible.)